frontend/ui/widget/textviewer.lua

```diff
@@ -272,6 +274,16 @@ function TextViewer:init(reinit)
         show_parent = self,
     }
 
+    -- NT: add titlebar.left_button (hamburger menu) to FocusManager.
+    if Device:hasDPad() and not Device:useDPadAsActionKeys() then
```

Comment:
What's the purpose of the complex hard to follow conditions instead of just keeping it simple?

Reply:
In layman's terms, don't add it on kindle or other device where there might be a Menu button. Although come to think of it, it could also be ` hasDPad and hasFewKeys`

Reply:
But why not? Why make things difficult and complex?

Reply:
Because having things on older kindles that don’t really need to be there makes it slower, particularly k2 and k3. Yes it is just one extra item to cycle through, but that one extra key press takes time, furthermore reduces performance so, i’d rather not have it.

if i was selfish, i would only add the button mapping. But since i am in the business of making it better to all, i also want other to be able to have it.

Reply:
> Yes it is just one extra item to cycle through, but that one extra key press takes time, 

There's a fundamental flaw in the reasoning there. Initial focus should be somewhere decent. That's it. That's the desired result. Nothing more, nothing less.

Also note that #12328 looks nothing like this. While it smells like something that should be a user patch it's gated behind `Device:hasSymKey() or Device:hasScreenKB()`. Hopefully other Kindle users don't share my opinion on it, because the simple fact is *other Kindle users added it in the first place*.

Reply:
Yes, that was added to kindle specifically, albeit before anyone considered the more logical approach of mapping things to buttons. And i am sure nobody is going to be upset because they now need to press fewer buttons.

Now as for hasDPad and not useDPad, tht is very much what i am seeking counsel on, it doesn’t really need to be on kindle, as it is already there through the key event. So what is the best way of reaching everyone yet not adding it to where it should not go.

Reply:
> And i am sure nobody is going to be upset because they now need to press fewer buttons and in turn gain improvements in performance.

I am. I think it tanks usability having to press a menu button instead of just happily navigating. But no matter, don't you simply want `hasDPad and not (Device:hasSymKey() or Device:hasScreenKB())`?

Reply:
> I am. I think it tanks usability having to press a menu button instead of just happily navigating.

yes, I would agree with this if only those old kindles had newer and faster processors but alas they don't and that "free navigation" comes at a cost. I wish you could see it in person and then you would perhaps understand.

> But no matter, don't you simply want hasDPad and not (Device:hasSymKey() or Device:hasScreenKB())

that is perhaps what I want, in trying to be clever (or helpful) I might be overtaking this whole matter.